lua/nvim-tree/view.lua

```diff
@@ -130,16 +130,29 @@ local function open_window()
   set_window_options_and_buffer()
 end
 
-local function get_existing_buffers()
-  return vim.tbl_filter(function(buf)
-    return a.nvim_buf_is_valid(buf) and vim.fn.buflisted(buf) == 1
-  end, a.nvim_list_bufs())
+local function get_alt_or_next_buf()
+  local alt_bufnr = vim.fn.bufnr "#"
+
+  local alt_buf = nil
+  local next_buf = nil
+
+  for _, buf in ipairs(a.nvim_list_bufs()) do
+    if a.nvim_buf_is_valid(buf) and vim.fn.buflisted(buf) == 1 then
+      next_buf = next_buf or buf
+      if buf == alt_bufnr then
+        alt_buf = buf
+      end
+    end
+  end
+
+  return alt_buf or next_buf
```

Comment:
The logic could be greatly simplified:
- check at the start that alt_bufnr is valid and listed, and return it in that case (create a small function like is_buf_displayed or something like that).
- then do the loop and return whenever you find a valid buffer

Reply:
nice